### ISSUE TITLE
Sir can i apply this wavelet neural network using theano on Iris data set. I applied on Iris dataset it is giving very big test score error.can u please help me out for getting good accuracy using your code or atleast give any suggestions so that it would be helpful for me.

### ISSUE THREAD

**Issue by satishdoppalapudi** [ameybarapatre/WaveletNeuralNetwork#2]
epoch 827, minibatch 12/12, validation error 66.657396 %
     epoch 827, minibatch 12/12, test error of best model 66.666712 %
epoch 828, minibatch 12/12, validation error 66.657394 %
     epoch 828, minibatch 12/12, test error of best model 66.666711 %
epoch 829, minibatch 12/12, validation error 66.657392 %
     epoch 829, minibatch 12/12, test error of best model 66.666711 %
epoch 830, minibatch 12/12, validation error 66.657390 %
     epoch 830, minibatch 12/12, test error of best model 66.666711 %
epoch 831, minibatch 12/12, validation error 66.657388 %
     epoch 831, minibatch 12/12, test error of best model 66.666711 %
epoch 832, minibatch 12/12, validation error 66.657386 %
     epoch 832, minibatch 12/12, test error of best model 66.666711 %
epoch 833, minibatch 12/12, validation error 66.657384 %
     epoch 833, minibatch 12/12, test error of best model 66.666710 %
Optimization complete. Best validation score of 66.657384 % obtained at iteration 9996, with test performance 66.666710 %
0.66666710415

The code for file __file__ ran for 0.03m

I did one hot encoding for Iris data set.

**Comment by ameybarapatre** [comment]
1) Change the loss function  it is  mean squared error here, use cross-entropy for classification
2) Consider dimensionality reduction.
3) play around with no of iterations and learning rate to get proper convergence.
